galaxy_ng/app/api/v3/viewsets.py

```diff
@@ -1,310 +1,72 @@
-
-    def _fix_list_urls(self, data):
-        namespace = self.kwargs['namespace']
-        name = self.kwargs['name']
-
-        for item in data:
-            version = item['version']
-            item['href'] = reverse(
-                'galaxy:api:v3:collection-version',
-                kwargs=dict(namespace=namespace, name=name, version=version)
-            )
-
-    def _fix_retrieve_url(self, data):
-        namespace = self.kwargs['namespace']
-        name = self.kwargs['name']
-        version = self.kwargs['version']
-
-        data['href'] = reverse(
-            'galaxy:api:v3:collection-version',
-            kwargs=dict(namespace=namespace, name=name, version=version)
-        )
-        data['collection']['href'] = reverse(
-            'galaxy:api:v3:collection',
-            kwargs=dict(namespace=namespace, name=name)
-        )
-
-
-class CollectionImportViewSet(api_base.ViewSet):
-
-    def retrieve(self, request, pk):
-        api = galaxy_pulp.GalaxyImportsApi(pulp.get_client())
-        response = api.get(prefix=settings.X_PULP_API_PREFIX, id=pk)
-        return Response(response.to_dict())
-
-
-class CollectionArtifactUploadView(api_base.APIView):
+        """
+        Returns a CollectionVersion object.
+        """
+        instance = self.get_object()
+        artifact = ContentArtifact.objects.get(content=instance)
 
-    permission_classes = api_base.GALAXY_PERMISSION_CLASSES + [
-        permissions.IsNamespaceOwner
-    ]
+        context = self.get_serializer_context()
+        context["content_artifact"] = artifact
 
-    parser_classes = [AnsibleGalaxy29MultiPartParser]
+        serializer = self.get_serializer_class()(instance, context=context)
+        return Response(serializer.data)
 
-    def post(self, request, *args, **kwargs):
-        metrics.collection_import_attempts.inc()
-        serializer = CollectionUploadSerializer(data=request.data, context={'request': request})
-        serializer.is_valid(raise_exception=True)
 
-        data = serializer.validated_data
-        filename = data['filename']
+class CollectionImportViewSet(LocalSettingsMixin, pulp_ansible_views.CollectionImportViewSet):
+    pass
 
-        try:
-            namespace = models.Namespace.objects.get(name=filename.namespace)
-        except models.Namespace.DoesNotExist:
-            raise ValidationError(
-                'Namespace "{0}" does not exist.'.format(filename.namespace)
-            )
 
-        self.check_object_permissions(request, namespace)
-        api = pulp.get_client()
-        url = '{host}/{prefix}/{path}'.format(
-            host=api.configuration.host,
-            path='v3/artifacts/collections/',
-            prefix=settings.X_PULP_API_PREFIX
-        )
-        headers = {}
-        headers.update(api.default_headers)
-        headers.update({'Content-Type': 'multipart/form-data'})
+class CollectionUploadViewSet(LocalSettingsMixin, pulp_ansible_views.CollectionUploadViewSet):
```

Comment:
If we need to replace `CollectionArtifactUploadView` with `CollectionUploadViewSet` which subclasses, `pulp_ansible_views.CollectionUploadViewSet` we'll need to find a way to keep with existing workflow. 

With this change it seems we no longer create a `galaxy_ng.app.models.CollectionImport` so we don't create import task messages and cannot show imports in a list. It seems this creates a `pulp_ansible.app.models.CollectionImport` instead.

Reply:
Yeah, going to need to implement the galaxy_collectionimport record generation in the CollectionUploaderViewSet create here.